include/net/mqtt.h

```diff
@@ -715,6 +715,20 @@ int mqtt_read_publish_payload(struct mqtt_client *client, void *buffer,
 int mqtt_read_publish_payload_blocking(struct mqtt_client *client, void *buffer,
 				       size_t length);
 
+/**
+ * @brief Waitall version of @ref mqtt_read_publish_payload function.
+ *
+ * @param[in] client Client instance for which the procedure is requested.
+ *                   Shall not be NULL.
```

Comment:
Perhaps "Must not be NULL." here instead

Reply:
> Perhaps "Must not be NULL." here instead

This is consistent with "shall not" as sprinkled originally around by @rlubos. IMHO, would be rather removed at all (Of course it can't be NULL! Also can't be 0x1 and many other values.)

Reply:
Kept as is to be consistent with other docs in this file.